### ISSUE TITLE
`uv pip --python` doesn't recognise / as a path separator on Windows

### ISSUE THREAD

**Issue by pfmoore** [astral-sh/uv#3062]
To demonstrate:

```
❯ uv venv a
Using Python 3.12.2 interpreter at: C:\Users\Gustav\AppData\Local\Programs\Python\Python312\python.exe
Creating virtualenv at: a
Activate with: overlay use a\Scripts\activate.nu
❯ uv pip list --python a/Scripts/python.exe
error: Failed to locate Python interpreter at `a/Scripts/python.exe`
❯ uv pip list --python a\Scripts\python.exe
```

Problem is here: https://github.com/astral-sh/uv/blob/main/crates/uv-interpreter/src/find_python.rs#L46

The line indicated only handles the "main" separator (`\` on Windows). But `/` is a perfectly valid path separator on Windows...

**Comment by charliermarsh** [comment]
Thanks, will fix.

**Comment by pfmoore** [comment]
A further issue caused by this is that if I chdir into the `Scripts` directory of a venv, and run `uv pip list --python python.exe` it will search for `python.exe` on `PATH` rather than using the venv. So, `uv pip list --python .\python.exe` gives a different answer than `uv pip list --python python.exe`, which is at best weird, and personally I'd consider it incorrect.

**Comment by zanieb** [comment]
I believe that problem would be addressed by the design at #2386 but I'd have to double check. Agree it's weird..

**Comment by charliermarsh** [comment]
@zanieb - My brief skim is that it's slightly separate and could even be changed / implemented before we do #2386 (e.g. via what's proposed at https://github.com/astral-sh/uv/issues/3060#issuecomment-2059301772).

**Comment by charliermarsh** [comment]
I'll fix this specific failing (the separator detection) and the rest let's track in https://github.com/astral-sh/uv/issues/2386#issuecomment-2059345905.

**Comment by charliermarsh** [comment]
I believe this was also fixed by #3064.
